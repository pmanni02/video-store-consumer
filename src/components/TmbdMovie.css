.tmbd-movie {
  margin: 2em;
  padding: 1.5em;
  width: 15em;
  background-color: #C7C6FF;
}

.tmbd-movie img {
  padding-top: 1em;
}
