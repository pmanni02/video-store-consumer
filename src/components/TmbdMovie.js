import React, { Component } from 'react';
import PropTypes from 'prop-types';

class TmbdMovie extends Component {
  constructor(props){
    super(props);
  }

  imageURL = (imageId) => {
    const imgPath = `https://image.tmdb.org/t/p/w300${imageId}`
    // console.log(imgPath);
    return imgPath;
  }

  onFormSubmit = (event) => {
    event.preventDefault();
    console.log(event.target);
    // ideally a popup for confirmation of movie selected
    // check if movie is already in videostore api
    // if not, make a POST? request to the videostore api
    // show status message for success or failure
    // render Rental component
  }

  render(){
    return(
      <form onSubmit = {this.onFormSubmit}>
        <p> Title: {this.props.title} </p>
        <p>Release Date: {this.props.releaseDate}</p>
        <img
          src = {this.imageURL(this.props.poster)}
          alt='movie poster'/>
        <input type = "submit" value = "Add Movie"/>
      </form>
    );
  }
}

TmbdMovie.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  releaseDate: PropTypes.string,
  poster: PropTypes.string
};


export default TmbdMovie;
