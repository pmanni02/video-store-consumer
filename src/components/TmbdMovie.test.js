import React from 'react';
import { shallow } from 'enzyme';
import TmbdMovie from './TmbdMovie';

describe('TmbdMovie', () => {
  test('snapshot', () => {
    const wrapper = shallow(
      <TmbdMovie
        id= {0}
        title=''
        releaseDate=''
        poster=''
        overview=''
        statusCallback={()=>{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
