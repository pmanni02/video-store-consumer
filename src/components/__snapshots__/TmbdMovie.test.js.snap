// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmbdMovie snapshot 1`] = `
<form
  className="tmbd-movie"
  onSubmit={[Function]}
>
  <h2>
     
  </h2>
  <p>
    Release Date: 
  </p>
  <img
    alt="movie poster"
    src=""
  />
  <button
    type="submit"
  >
    Add Movie
  </button>
</form>
`;
